### PR TITLE
Specify ISO8601Calendar.

### PR DESCRIPTION
**Pull request by Ms2ger**
This leaves the actual algorithms undefined, but it serves as a skeleton for
the work that needs to be done.

### CODE REVIEW

**Review by ptomato — APPROVED** [review]
LGTM! Up to you whether you prefer to merge it in advance of the discussion being reopened on #847 or not. I think it's fine to merge it since it does reflect what's currently in the polyfill.

File: polyfill/lib/calendar.mjs [hunk 1]
```diff
@@ -114,7 +114,7 @@ export class Calendar {
 MakeIntrinsicClass(Calendar, 'Temporal.Calendar');
 DefineIntrinsic('Temporal.Calendar.from', Calendar.from);
 
-class ISO8601 extends Calendar {
+class ISO8601Calendar extends Calendar {
```
Comment (ryzokuken) [inline polyfill/lib/calendar.mjs:117]: Why this change when it's `Japanese` and not `JapaneseCalendar`?

**Review by ryzokuken — APPROVED** [review]
LGTM, thanks.

File: polyfill/lib/calendar.mjs (hunk 1 above)
Comment (Ms2ger) [reply polyfill/lib/calendar.mjs:117]: I haven't looked at the Japanese calendar yet.

Comment (ryzokuken) [reply polyfill/lib/calendar.mjs:117]: Basically, my question is, do you plan to suffix each class by `Calendar`? It looks weird when it's inconsistent, but if you plan to eventually rename all of them, then it's all good.

Comment (Ms2ger) [reply polyfill/lib/calendar.mjs:117]: If we end up going with subclassing, I think suffixing them all makes sense, yes.

**Comment by ryzokuken** [comment]
Going ahead with the merge because https://github.com/tc39/proposal-temporal/pull/879#discussion_r485441277
